discojs/src/models/gpt/index.ts

```diff
@@ -122,17 +112,45 @@ export class GPT extends Model {
     };
   }
 
-  override predict(input: Sample): Promise<Prediction> {
-    const ret = this.model.predict(input);
-    if (Array.isArray(ret)) {
-      throw new Error(
-        "prediction yield many Tensors but should have only returned one",
-      );
-    }
+  #batchToTF(batch: Batched<ModelEncoded["text"]>): {
+    xs: tf.Tensor2D;
+    ys: tf.Tensor3D;
+  } {
+    return tf.tidy(() => ({
+      xs: tf.stack(
+        batch.map(([line]) => tf.tensor1d(line.toArray(), "int32")).toArray(),
+      ) as tf.Tensor2D, // cast as stack doesn't type
+      ys: tf.stack(
+        batch
+          .map(([line, next]) =>
+            tf.oneHot(line.shift().push(next).toArray(), this.#vocabSize),
+          )
+          .toArray(),
+      ) as tf.Tensor3D, // cast as oneHot/stack doesn't type
+    }));
+  }
+
+  override async predict(
+    batch: Batched<ModelEncoded["text"][0]>,
+  ): Promise<Batched<ModelEncoded["text"][1]>> {
+    const predictNext = async (tokens: List<number>) => {
+      const generated = await this.model.generate(tokens.toArray(), {
+        maxNewTokens: 1,
+        temperature: 1.0,
+        doSample: false,
+      });
```

Comment:
well, that's a good point. I'm trying to think of a generic way to allow predict to do that.
* having it as the `Model` level doesn't work as every implementation might have a different config (temperature for 'tabular' models?)
  * note: if we find a config usable across a given DataType, we can have that instead
* having it as the implementation (GPT/TFJS) level looks right, with an extra argument to `predict` which is only valid after a `instanceof` check

I'm having an issue with returning many tokens from GPT, as that would break the types (a single token is used as output in `ModelEncoded`). I would argue that it's weird to ask for many tokens. IMO, in a loop, one should ask for the next one and append it to the input (as it is already done in `GPTForCausalLM.generate`).

after losing myself in code, I choose to only have a single token generated, and to expose the config in GPT direclty. that put a bit a burden on the users to pre/post-process the lines, but `Model` isn't made for users.
at some later point, I was thinking of merging `Validator` into `Disco` (as to have a single entry-point), and thus to provide a nice predict method there, with a selectable number of token.

tl;dr no, it wasn't intended, put it back but nicer